controllers/apply/form-router-next/joi-extensions/compare-object.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+const isEqual = require('lodash/isEqual');
+
+module.exports = function(joi) {
+    return {
+        base: joi.object(),
+        name: 'object',
+        language: {
+            isEqual: 'Objects must not match'
+        },
+        rules: [
+            {
+                name: 'compare',
+                params: {
+                    ref: joi.func().ref()
+                },
+                validate(params, value, state, options) {
+                    const refVal = params.ref(
```

Comment:
💯 

I was [trying to figure this out](https://stackoverflow.com/questions/56689438/comparing-two-objects-in-joi-validation-eg-to-avoid-duplicates) and couldn't, but *knew* there'd be a smart way to get the reference value. Nice job! 👍